src/main/java/com/software/modsen/passengermicroservice/services/PassengerServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.software.modsen.passengermicroservice.services;
+
+import com.software.modsen.passengermicroservice.entities.Passenger;
+import com.software.modsen.passengermicroservice.entities.PassengerDto;
+import com.software.modsen.passengermicroservice.entities.PassengerPatchDto;
+import com.software.modsen.passengermicroservice.exceptions.ErrorMessage;
+import com.software.modsen.passengermicroservice.exceptions.PassengerNotFoundException;
+import com.software.modsen.passengermicroservice.exceptions.PassengerWasDeletedException;
+import com.software.modsen.passengermicroservice.mappers.PassengerMapper;
+import com.software.modsen.passengermicroservice.repositories.PassengerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class PassengerServiceImpl implements PassengerService {
+    @Autowired
+    private PassengerRepository passengerRepository;
+    private final PassengerMapper PASSENGER_MAPPER = PassengerMapper.INSTANCE;
+
+    public Passenger getPassengerById(long id) {
+        Optional<Passenger> passengerFromDb = passengerRepository.findById(id);
+        if (passengerFromDb.isPresent()) {
+            if (!passengerFromDb.get().isDeleted()) {
+                return passengerFromDb.get();
+            }
+
+            throw new PassengerWasDeletedException(ErrorMessage.PASSENGER_WAS_DELETED_MESSAGE);
+        }
```

Comment:
You can use method of Optional class like ifPresent(passenger -> {action})

Reply:
Refactor